src/scripts/destinyTrackerApi/bulkFetcher.js

```diff
@@ -0,0 +1,81 @@
+import { itemListBuilder } from './itemListBuilder.js';
+
+class bulkFetcher {
+  constructor($q, $http, trackerErrorHandler, loadingTracker, scoreMaintainer) {
+    this.$q = $q;
+    this.$http = $http;
+    this._itemListBuilder = new itemListBuilder();
+    this._trackerErrorHandler = trackerErrorHandler;
+    this._loadingTracker = loadingTracker;
+    this._reviewDataCache = scoreMaintainer;
+  }
+
+  getBulkWeaponDataPromise(gunList) {
+    return {
+      method: 'POST',
+      url: 'https://reviews-api.destinytracker.net/api/weaponChecker/fetch',
+      data: gunList,
+      dataType: 'json'
+    };
+  }
+
+  getBulkFetchPromise(stores) {
+    if (stores.stores.length === 0) {
```

Comment:
You can use `if (!stores.stores.length)` here like you do below.

You might want to document what `stores` looks like because `stores.stores` is confusing.